Exceptions from aGrUM
=====================

Exception in pyAgrum
--------------------

All the exception classes inherit pyagrum.GumException's functions.

.. autoexception:: pyagrum.GumException

List of pyAgrum's exceptions
----------------------------

.. autoexception:: pyagrum.DefaultInLabel
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.DuplicateElement
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.DuplicateLabel
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.FatalError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.FormatNotFound
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.GraphError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.IOError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidArc
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidArgument
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidArgumentsNumber
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidDirectedCycle
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidEdge
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.InvalidNode
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.NoChild
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.NoNeighbour
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.NoParent
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.NotFound
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.NullElement
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.OperationNotAllowed
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.OutOfBounds
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.ArgumentError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.SizeError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.SyntaxError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.UndefinedElement
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.UndefinedIteratorKey
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.UndefinedIteratorValue
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.UnknownLabelInDatabase
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.DatabaseError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
.. autoexception:: pyagrum.CPTError
	:exclude-members: errorCallStack, errorContent, errorType, with_traceback, args, thisown, add_note, what
